app/deliver_grant_funding/routes.py

```diff
@@ -98,30 +104,47 @@ def grant_setup_contact() -> ResponseReturnValue:
     form = GrantContactForm(obj=grant_session)
 
     if form.validate_on_submit():
-        try:
-            assert form.primary_contact_name.data, "Primary contact name must be provided"
-            assert form.primary_contact_email.data, "Primary contact email must be provided"
-
-            grant_session.primary_contact_name = form.primary_contact_name.data
-            grant_session.primary_contact_email = form.primary_contact_email.data
+        assert form.primary_contact_name.data, "Primary contact name must be provided"
+        assert form.primary_contact_email.data, "Primary contact email must be provided"
+        grant_session.primary_contact_name = form.primary_contact_name.data
+        grant_session.primary_contact_email = form.primary_contact_email.data
+        session["grant_setup"] = grant_session.to_session_dict()
+        return redirect(url_for("deliver_grant_funding.grant_setup_check_your_answers"))
 
-            grant = interfaces.grants.create_grant(
-                name=grant_session.name,
-                description=grant_session.description,
-                primary_contact_name=grant_session.primary_contact_name,
-                primary_contact_email=grant_session.primary_contact_email,
-                ggis_number=grant_session.ggis_number,
-            )
+    return render_template("deliver_grant_funding/grant_setup/contact.html", form=form)
 
-            session.pop("grant_setup", None)
 
-            return redirect(url_for("deliver_grant_funding.view_grant", grant_id=grant.id))
+@deliver_grant_funding_blueprint.route("/grants/setup/check-your-answers", methods=["GET", "POST"])
+@platform_admin_role_required
+@auto_commit_after_request
+def grant_setup_check_your_answers() -> ResponseReturnValue:
+    if "grant_setup" not in session:
+        return redirect(url_for("deliver_grant_funding.grant_setup_intro"))
 
-        except DuplicateValueError as e:
-            field_with_error: Field = getattr(form, e.field_name)
-            field_with_error.errors.append(f"{field_with_error.name.capitalize()} already in use")  # type:ignore[attr-defined]
+    grant_session = GrantSetupSession.from_session(session["grant_setup"])
+    form = GrantCheckYourAnswersForm()
 
-    return render_template("deliver_grant_funding/grant_setup/contact.html", form=form)
+    if form.validate_on_submit():
+        grant = interfaces.grants.create_grant(
+            name=grant_session.name,
+            description=grant_session.description,
+            primary_contact_name=grant_session.primary_contact_name,
+            primary_contact_email=grant_session.primary_contact_email,
+            ggis_number=grant_session.ggis_number,
+        )
+        session.pop("grant_setup", None)
+        return redirect(url_for("deliver_grant_funding.grant_setup_confirmation", grant_id=grant.id))
```

Comment:
As you've flagged, we don't do any error handling here on things like the grant name unique constraint. We do have that on the grant name form page though.

I am broadly OK with this; it will be a low traffic page and the chance of meaningful errors that occur here but not on the original grant name page should be low.

If we start getting errors from this page in sentry, we can rethink.